homeassistant/components/generic/camera.py

```diff
@@ -12,6 +12,7 @@
     Camera,
     CameraEntityFeature,
 )
+from homeassistant.components.stream import CONF_RTSP_TRANSPORT, RTSP_TRANSPORTS
```

Comment:
My impression is there is a preference to import this from .const rather than re-import from stream which has this because its imported from const.

Reply:
Do you have a source for this? I vaguely remember a conversation about this (exporting constants through __init__.py) in the discord several months ago but I don't recall the details. Just want to be sure which convention is preferred.

Reply:
I was thinking of PRs like this https://github.com/home-assistant/core/pull/63469 -- @cdce8p may be able to better articulate the convention if there is one?

Reply:
I found the conversation...look on the discord from Jan 21...

Reply:
IMO it's fine if these integrations depend on `components.stream` anyway.
@allenporter The PR you linked was more about constants also defined in `homeassistant.const`. Those should obviously be imported from there and not from another integration.

There was also a related discussion about weather the import should be from, in this case, `homeassistant.components.stream` or `homeassistant.components.stream.const`. #63820
Mainly because of the `no-implicit-reexport` mypy rule, which isn't enforced for integrations yet though. The result of that discussion was other integrations should import from `stream` directly and `stream/__init__.py` should provide a `__all__` constant to export these explicitly. https://github.com/home-assistant/core/pull/63820#issuecomment-1019502344

Tldr, the current approach is probably fine. At some point `__all__` should be added to `stream/__init__.py` though.

Reply:
OK -- `no-implicit-reexport` is what i was thinking of. thanks!

Reply:
Thanks for the info. How about I import these from stream.const for now for consistency, then we follow up with another PR which moves all exports into `__init__.py`  / `__all__`